files-override/ts/app/app.ts

```diff
@@ -0,0 +1,25 @@
+import Application from '@ember/application';
+// This can be removed if you have no need for compatModules
+// @ts-expect-error TODO: add types to @embroider/core/entrypoint
```

Comment:
I feel like instead of having a difference here we should just fix it upstream 🤔 we know we need to provide a d.ts file for the entrypoint and it's going to be a simple change

Reply:
yes, but I wanted to make progress.
when types are added tests will break, and then I can fix 🎉

Reply:
> when types are added tests will break

is that because you are telling TS that it expects an error and if it doesn't find one it fails?

Reply:
correct